test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -154,6 +169,39 @@ def check_output(self):
       "hwaccel initialisation returned error", self.output, re.MULTILINE)
     assert m is None, "Failed to use hardware decode"
 
+  # scale scaled test stream to source file width and height for generating yuv
+  def scale_to_source_size(self, scaled, codec, mode, width, height, frames, enc_file):
+    _, _, ffdecparms = self.get_requirements_data("decode", codec, mode)
+    ffdeccodec, _ = ffdecparms
+    _, _, ffencparms = self.get_requirements_data("encode", codec, mode)
+    ffenccodec, ext = ffencparms
+    # TBD - for mjpeg and mpeg2, HW scaling not working so using SW scaling
+    # Remove this once HW scaling works for jpeg and mjpeg2
+    if ext == "mjpeg":
+      opts = " -c:v mjpeg"
+    elif ext == "m2v":
+      opts = " -c:v mpeg2video"
+    else:
+      opts = " -hwaccel qsv -hwaccel_device /dev/dri/renderD128 -init_hw_device qsv=hw -c:v {} ".format(ffdeccodec)
+
```

Comment:
You decode+scale to coded file in one call, then decode (the scaled coded file) to yuv in another call.  This is unnecessary.  It would be better to decode+scale directly to yuv in a single call.

Reply:
implemented decode+scale yuv generation but seeing some issues with multi codec scaling